src/main/java/lotto/conotroller/Lotto.java

```diff
@@ -0,0 +1,26 @@
+package lotto.conotroller;
+
+import lotto.domain.AutoNumberGenerator;
+import lotto.domain.LottoLottery;
+import lotto.domain.LottoPurchaseAmount;
+import lotto.domain.LottoQuantity;
+import lotto.domain.WinningNumbers;
+import lotto.domain.WinningRanks;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class Lotto {
+    public void run() {
+        LottoPurchaseAmount lottoPurchaseAmount = new LottoPurchaseAmount(InputView.inputPurchaseAmount());
+        LottoQuantity lottoQuantity = LottoQuantity.of(lottoPurchaseAmount.getQuantity());
+        OutputView.println(lottoQuantity.getMessage());
+
+        LottoLottery lottoLottery = LottoLottery.of(lottoQuantity, new AutoNumberGenerator());
```

Comment:
`LottoPurchaseAmount` 구매수량을 결정하는 객체인것 같습니다.
그렇다면, 구매수량(int)를 가지고 와서 외부에서 판단하는게 아닌 객체에게 물어보면 어떨까요?

```suggestion
        LottoQuantity lottoQuantity = lottoPurchaseAmount.toLottoQuantity();
```

- 구매수량(int)를 주세요.
  - 데이터와 역할 or 책임의 응집성이 떨어질 수 있음, 가져가는 곳에서 판단하기때문에. ​

- 구매수량(LottoQuantity)이 무엇인가요?
  - `private final int amount;` private으로 지정했지만, `getter`로 인해 데이터의 캡슐화의 의미가 퇴색이 된것 같아요.
  - 데이터와 기능의 응집성이 높아진다고 생각합니다.


--- 

무조건의 `getter`를 지양해야한다는것은 아닙니다! 
정답은 없습니다!!
다른 부분도 전체적으로 한번 고민해봐주세요~

[TellDontAsk](https://martinfowler.com/bliki/TellDontAsk.html)

Reply:
```suggestion
        LottoLottery lottoLottery = lottoPurchaseAmount.toLottoLottery();
```
Suggeste 가 잘못되었네요 ^^;; 수정합니다.

Reply:
음.. getQuantity라고해서 getter처럼 보였나보네요 calculateQuantity라고 해야 좀 더 명확해 질까 싶은데..
lottoPurchaseAmount는 구매하기 위해 입력한 금액이고,
그걸 통해 quantity를 계산해서 로또 구매 수량인 LottoQuantity를 생성하고,
LottoLottery는 로또 복권으로 로또 1장이라고 보시면 될거같습니다.
구매금액을 통해 구매 수량을 알았으니 LottoLottery 로또를 갯수만큼 로또번호생성 방법에따라 생성한다는 의미였는데 일단 메소드명을 명확하게 변경하겠습니다
getter 자체를 생성 안하려고 노력했었는데 getter로 생각하신 것 같습니다.

Reply:
말씀하신 부분 이해가 된건지 모르겠습니다.. 일단 변경해 봣는데 조언해 주신 부분이 적용된건지 확인 부탁드리겠습니다 !
어렵네요..ㅎㅎㅎㅎ

Reply:
TDD 개발 패턴과 DDD를 경험해보고 익혀보기 위함 과정이라고 생각해주시면 좋을것같아요!
해당 부분에 대해서 많이 찾아보고 다양한 시도를 해보시는것을 추천드려요!
과정에서 궁금하거나 안풀리는 부분이 있으면 DM으로 같이 고민해보면 좋을것 같아요 :+1: